objectivec/ort_value_internal.h

```diff
@@ -9,6 +9,14 @@ NS_ASSUME_NONNULL_BEGIN
 
 @interface ORTValue ()
 
+/**
+ * Creates a value that wraps an existing C API OrtValue and takes ownership of it.
```

Comment:
The message might need to be more explicit about what happens on failure. Personally I would have expected to still own the CApiOrtValue and for it to not be released if the attempt to create a wrapper type failed so I can choose if I want to release at that point or not. Either approach is fine - but the doco should call this out.  

FWIW doesn't the existing code here to expect to release the CApiOrtValue on failure and should be updated to set `j = i+1` if we're going to release in initWithCAPIOrtValue?

https://github.com/microsoft/onnxruntime/blob/eb4e6d206233a7a62c3f238cf2edea2b37684719/objectivec/ort_session.mm#L125-L129

Reply:
good catch. I will try to make this easier to use by using `Ort::Value`'s and wrapping the `OrtValue`s at the call site. kind of tricky when dealing with the `OrtValue`s directly.